napari/layers/labels/labels.py

```diff
@@ -562,15 +561,34 @@ def _set_colormap(self, colormap):
     @property
     def num_colors(self):
         """int: Number of unique colors to use in colormap."""
-        return self._num_colors
+        warnings.warn(
+            trans._(
+                'Labels.num_colors is deprecated since 0.4.19 and will be '
+                'removed in 0.5.0, please use len(Labels.colormap) - 1 '
+                'instead.',
+                deferred=True,
+            ),
+            FutureWarning,
+            stacklevel=2,
+        )
+        return len(self.colormap)
 
     @num_colors.setter
     def num_colors(self, num_colors):
+        warnings.warn(
+            trans._(
+                'Setting Labels.num_colors is deprecated since 0.4.19 and '
+                'will be removed in 0.5.0, please set Labels.colormap '
+                'instead.',
```

Comment:
Maybe point to `napari.utils.colormaps.label_colormap` function?

Reply:
I think I want to discourage/remove that also in the future — we should be able to use the LabelColormap constructor with a list of colors. Maybe we make a `low_discrepancy_colors` function to generate it and then it's `LabelColormap(low_discrepancy_colors(5))`. What do you think?

Reply:
ok

Reply:
And such a function should be part of this PR to allow switching to the new API.

Reply:
So what here?